### PR TITLE
Bigfix Issue #23

### PR DESCRIPTION
**Pull request by Razumain**
Fixing bug described in Issue #23

### CODE REVIEW

File: commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java [hunk 1]
```diff
@@ -302,13 +302,39 @@ public static String getAlgorithmURI(final ASN1ObjectIdentifier sigAlgoOid, fina
...
-      .filter(algoProp -> algoProp.getSigAlgoOID().equals(sigAlgoOid) &&
+      .filter(algoProp -> isSigAlgoEquivalent(algoProp.getSigAlgoOID(), sigAlgoOid, digestAlgoOid) &&
           algoProp.getDigestAlgoOID().equals(digestAlgoOid))
       .map(PDFSignatureAlgorithmProperties::getSigAlgoId)
       .findFirst()
       .orElseThrow(() -> new NoSuchAlgorithmException("Non supported combination of signature algorithm and hash algorithm"));
   }
 
+  /**
+   * This function is designed to allow identifiers for RSA encryption to be equivalent to identifiers for various RSA combined with various hash functions
```
Comment (martin-lindstrom) [inline commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java:313]: function -> method

File: commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java [hunk 2]
```diff
@@ -302,13 +302,39 @@ public static String getAlgorithmURI(final ASN1ObjectIdentifier sigAlgoOid, fina
...
+   * @param sigAlgoPropOID
```
Comment (martin-lindstrom) [inline commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java:314]: Missing documentation for param and return

File: commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java [hunk 3]
```diff
@@ -302,13 +302,39 @@ public static String getAlgorithmURI(final ASN1ObjectIdentifier sigAlgoOid, fina
...
+   * @param cmsSigAlgoOID
+   * @return
+   */
+  private static boolean isSigAlgoEquivalent(ASN1ObjectIdentifier sigAlgoPropOID, ASN1ObjectIdentifier cmsSigAlgoOID, ASN1ObjectIdentifier cmsDigestAlgoOID) {
```
Comment (martin-lindstrom) [inline commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java:318]: Make parameters final

**Review by martin-lindstrom — CHANGES_REQUESTED** [review]
Some minor changes required.

**Comment by Razumain** [comment]
The latest update should resolve all comments

File: commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java (hunk 1 above)
Comment (Razumain) [reply commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java]: Fixed

File: commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java (hunk 2 above)
Comment (Razumain) [reply commons/pdf-commons/src/main/java/se/idsec/signservice/security/sign/pdf/configuration/PDFAlgorithmRegistry.java]: Fixed

**Pull request merged**